examples/plot_compare_reduction.py

```diff
@@ -72,4 +86,45 @@
 plt.ylabel('Digit classification accuracy')
 plt.ylim((0, 1))
 plt.legend(loc='upper left')
+
+###############################################################################
+# Caching transformers within a ``Pipeline``
```

Comment:
Actually this example does not benefit either because the fit is too quick and pickling / unpickling large dicts with string keys (the vocabulary of the vectorizers) is comparatively more expensive.